pdns/dnsdist.cc

```diff
@@ -1403,6 +1403,12 @@ static void processUDPQuery(ClientState& cs, LocalHolders& holders, const struct
       return;
     }
 
+    if (queryHasEDNS(dq) && getEDNSVersion(dq) > 0) {
+        dq.dh->qr = true;
+        dq.dh->rcode = (16 & 0xF);
+        dq.ednsRCode = ((16 & 0xFFF0)>>4); // set rcode to BADVERS
```

Comment:
I'd really prefer the use of `ERCode::BADVERS` over `16`. I'm also not sure why you set the "regular" `rcode` here.